### PR TITLE
Do not remove existing title elements, assert instead

### PR DESCRIPTION
**Pull request by raido**
We no longer need to remove existing title elements, because we are setting `document.title` directly.

This effectively fixes also flicker issue mention in #184 and brings visibility of conflicting updates on dev time due to additional assertion.

Closes: #184

### CODE REVIEW

File: addon/services/page-title-list.js [hunk 1]
```diff
@@ -32,9 +37,7 @@ export default class PageTitleListService extends Service {
         }
       });
     }
-    this.router.on('routeDidChange', () => {
-      this.scheduleTitleUpdate();
-    });
+    this.router.on(RouterEvent.ROUTE_DID_CHANGE, this.scheduleTitleUpdate);
```
Comment (raido) [inline addon/services/page-title-list.js:40]: This is change is not directly related to #184 .

File: addon/services/page-title-list.js [hunk 2]
```diff
@@ -213,6 +216,11 @@ export default class PageTitleListService extends Service {
     return title.join('');
   }
 
+  willDestroy() {
+    super.willDestroy();
+    this.router.off(RouterEvent.ROUTE_DID_CHANGE, this.scheduleTitleUpdate);
```
Comment (raido) [inline addon/services/page-title-list.js:221]: In real life application service does not get destroyed during application session but in acceptance tests it is nice to clean up properly.

**Comment by knownasilya** [comment]
Nice!
